Products/DCWorkflow/tests/test_exportimport.py

```diff
@@ -998,6 +998,46 @@ def test_parseWorkflowXML_normal_permissions( self ):
 
             self.failUnless( permission in _WF_PERMISSIONS )
 
+    def test_parseWorkflowXML_normal_groups( self ):
+
+        WF_ID = 'normal'
+        WF_TITLE = 'Normal DCWorkflow'
+        WF_DESCRIPTION = 'Normal workflow'
+        WF_INITIAL_STATE = 'closed'
+
+        site = self._initSite()
+
+        configurator = self._makeOne( site ).__of__( site )
+
+        ( workflow_id
+        , title
+        , state_variable
+        , initial_state
+        , states
+        , transitions
+        , variables
+        , worklists
+        , permissions
+        , groups
+        , scripts
+        , description
+        , manager_bypass
+        , creation_guard
+        ) = configurator.parseWorkflowXML(
+                          _NORMAL_WORKFLOW_EXPORT
+                          % { 'workflow_id' : WF_ID
+                            , 'title' : WF_TITLE
+                            , 'description' : WF_DESCRIPTION
+                            , 'initial_state' : WF_INITIAL_STATE
+                            , 'workflow_filename' : WF_ID.replace(' ', '_')
+                            } )
+
+        self.assertEqual( len( groups ), len( _WF_GROUPS ) )
+
+        for group in groups:
+
+            self.failUnless( group in _WF_GROUPS )
```

Comment:
Two points:

- `failUnless` is deprecated (this really only matters for Py3k compatibility, where it would cause warnings, but we aren't there yet).  I realize there are probably dozens/hundreds of them in the products tests.

- You can simplify the assertion + loop into a set comparison:

```python
        self.assertEqual(set(groups), set(_WF_GROUPS))
```

If you use that, you don't need to fix the `failUnless` :).